add aap-setup-download role to download automatically the AAP setup file

ericzolf opened the pull request:
### What does this PR do?

Adds a role to automatically download the installer image from RHN

### How should this be tested?

My playbook looks so far like this:

```yaml
- name: download and install AAP from the bastion
  hosts: bastion.*
  gather_facts: false
  become: true
  tags: aap_installation
  roles:
    - redhat_cop.tower_utilities.aap_setup_download
  post_tasks:
    - debug:
        var: aap_setup_down_installer_file
```
And the only variable I really need to define in my vault is aap_setup_down_offline_token (see `defaults/main.yml` for the details and other variables).

### Is there a relevant Issue open for this?

n/a

### Other Relevant info, PRs, etc.

Adapted from https://www.redhat.com/en/blog/automating-installation-ansible-automation-platform-ansible-and-satellite

I still need to work on the "soft" aspects of the role, means documentation :-) hence still WiP, else it works good enough for me.

ericzolf committed:
Add aap-setup-download role to download automatically the AAP setup file

Adapted from https://www.redhat.com/en/blog/automating-installation-ansible-automation-platform-ansible-and-satellite

ericzolf committed:
Fix ansible-lint issues

ericzolf committed:
Write proper README for aap_setup_download

ericzolf commented:
Ready for review, I'm working currently on the follow-up roles `aap_setup_prepare` and `aap_setup_install` but it takes longer to properly test...

sean-m-sullivan commented:
I like the idea of this, and want to run it through some testing. However I think long term we need to incorporate the 
https://galaxy.ansible.com/middleware_automation/redhat_csp_download
plugin and/or role. As that does a lot of our lifting in a reproducable way.

sean-m-sullivan commented on roles/aap_setup_download/README.md:
```diff
@@ -0,0 +1,55 @@
+
+The following input variables are required:
+
+* `aap_setup_down_offline_token` contains your offline token as described in the requirements.
```
we should point the user in the right place to get the token, rather then going to the blog post. 
https://access.redhat.com/management/api is the link to follow.

sean-m-sullivan commented on roles/aap_setup_download/tasks/main.yml:
```diff
@@ -0,0 +1,38 @@
+      client_id: rhsm-api
+      refresh_token: "{{ aap_setup_down_offline_token }}"
+  register: __aap_setup_down_login
+- name: collecting the available installers
```
Should be a line between this task and the previous task.

sean-m-sullivan reviewed (changes requested):
This works well, And I left the previous comment about the module and looked into it, its not listed on that release page, and this eliminates the need to look up and ID number that module would require, even if it were on that page. 
With the documentation change and the one lint/style change I think this would be good to go!

sean-m-sullivan commented:
Upon further testing, and trying to integrate to a new version of our prereq role, I added a task of setting the filename as a variable, along with the other changes, as a PR to your branch before we merge this branch to devel.

ericzolf committed:
Address points raised by Sean

- link to location for API tokens added to readme
- missing blank line between tasks (is it actually our best practice?)
